scripts/new_upstream_snapshot.py

```diff
@@ -0,0 +1,627 @@
+#!/usr/bin/python3
+# Python 3.6+
```

Comment:
Is there any reason why you would not make this named 'new-upstream-snapshot'? its just a strange command name. 

Some justification:
1. you break anything (maybe it is just humans) that invoke new-upstream-snapshot
2. very few programs in my /usr/bin have a '.' in them at all, and even fewer have an extension that indicates the programming language they're written in.

Reply:
Point 2 is in line with https://www.debian.org/doc/debian-policy/ch-files.html#scripts.

Reply:
I tend to take the policy as a best practices guide even outside packaging, but yes no need to really concern ourselves with it in this case.

Reply:
> you break anything (maybe it is just humans) that invoke new-upstream-snapshot

In the current state I see this as possibly a good thing. The new script has different flags and while the PR says "including all of the original functionality", minor (and some maybe not-so-minor) inconsistencies typically crop up from wholesale rewrites like this.

Breaking early (at invocation), should help unknowing users realize that "something big changed" so they can verify functionality for their use case(s). If we had a larger user base I would probably carry a different tune regarding cli stability / etc, but I don't think it's too unreasonable here.

Reply:
This PR should definitely also update the docs in `doc/ubuntu_release_process.md` 

We could provide a symlink new-upstream-snapshot to this callable python module to make it easier on the eyes for folks with uss-tableflip/scripts in their `$PATH` but I think then we just have two entries in path new-upstream-snapshot and new_upstream_snapshot.py.  Given that this shift represents a significant breaking change here, it is going to force scripts or humans to slightly adapt their use of this script anyway. 

The following options are changed:
  - --sru-bug -> --bug

The following options are DROPPED:
  - -n, --no-sru-bug
  - --skip-branch-name-check
  - --no-bugs
  - --update-patches-only
  - -v